deepcell_tracking/utils.py

```diff
@@ -605,9 +611,19 @@ def get_image_features(X, y, appearance_dim=32):
     appearances = np.zeros((num_labels, appearance_dim,
                             appearance_dim, X.shape[-1]), dtype='float32')
 
+    # Zero-pad the X array for fixed crop mode
+    pad_width = ((appearance_dim, appearance_dim),
+                 (appearance_dim, appearance_dim),
+                 (0, 0))
+    X_padded = np.pad(X, pad_width=pad_width)
+    y_padded = np.pad(y, pad_width=pad_width)
+
```

Comment:
```suggestion
    if crop_mode == 'fixed':
        # Zero-pad the X array for fixed crop mode
        pad_width = ((appearance_dim, appearance_dim),
                     (appearance_dim, appearance_dim),
                     (0, 0))
        X_padded = np.pad(X, pad_width=pad_width)
        y_padded = np.pad(y, pad_width=pad_width)

        props = regionprops(y_padded[..., 0], cache=False)

```

This suggestion and the next are to avoid generating duplicate arrays and regionprops if its not needed